File: external-libs/core/src/main/java/com/serhii/core/security/Hash.java
package com.serhii.core.security;

import com.serhii.core.CoreEngine;
import com.serhii.core.security.impl.hash.HashGenerator;

public class Hash {

    private HashGenerator generator;

    public Hash() {
        CoreEngine.getInstance().configure(this);
    }

    public void setGenerator(HashGenerator generator) {
        this.generator = generator;
    }

    public String hashMD5(String message) {
        return generator.makeHashMD5(message);
    }
}

File: external-libs/android-core/src/main/java/com/serhii/core/security/Hash.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.core.security

import com.serhii.core.security.impl.hash.HashGenerator
import com.serhii.core.CoreEngine

/**
 * Hash APIs
 */
object Hash {
    fun hashMD5(message: String): String {
        val generator: HashGenerator = CoreEngine.configure(this)
        return generator.makeHashMD5(message)
    }
}